tests/BaseClient_Test.php

```diff
@@ -4,20 +4,46 @@
 use Recurly\Resources\TestResource;
 use Recurly\BaseClient;
 use Recurly\Utils;
+use Recurly\Logger;
+use Psr\Log\LogLevel;
 
 final class BaseClientTest extends RecurlyTestCase
 {
     public function setUp(): void
     {
         parent::setUp();
-        $this->client = new MockClient();
+        $logger = new Logger('Recurly', LogLevel::EMERGENCY);
+        $this->client = new MockClient($logger);
     }
 
     public function tearDown(): void
     {
         $this->client->clearScenarios();
     }
 
+    public function testDebugLoggerWarning(): void
+    {
+        $msg = <<<MSG
+        The Recurly logger should not be initialized
+        beyond the level INFO. It is currently configured to emit
+        headers and request \/ response bodies. This has the potential to leak
+        PII and other sensitive information and should never be used in production.
+        MSG;
+        $this->expectOutputRegex('/SECURITY WARNING: ' . $msg . '/');
+        $logger = new Logger('Recurly', LogLevel::DEBUG);
+        $client = new MockClient($logger);
+    }
+
+    public function testDefaultLogger(): void
+    {
+        $client = new MockClient(null);
+        $reflector = new ReflectionProperty('Recurly\BaseClient', '_logger');
```

Comment:
I don't like using reflection in tests to access private properties, but I couldn't formulate another way to easily verify the default logger was being set.

Reply:
Seems like a legitimate case to me. The alternative of observing output compatibility isn't strong enough of an assertion, and adding a `getLogger` would be inappropriate.